node/src/components/block_accumulator/block_acceptor.rs

```diff
@@ -144,6 +151,7 @@ impl BlockAcceptor {
         if false == had_sufficient_finality {
             if let Some(node_id) = peer {
                 self.register_peer(node_id);
+                check_signatures_from_peer_bound(validator_slots * 2, node_id, &self.signatures)?;
```

Comment:
Hmm, now when I think about it I see that just [above](https://github.com/casper-network/casper-node/pull/3709/files#diff-3b7f521fe50a87bc7200f3953b67b3961bd9761112f20611212bf026050e1b39L124) we have the expensive crypto verification of the signature.

Could it be better if we first check the bound, and if there are already too many signatures, we don't even invoke the crypto verification?

cc: @fizyk20

Reply:
Done